community/detectors/slurm_exposed_rest_api/src/test/java/com/google/tsunami/plugins/detectors/rce/SlurmExposedRestApiDaemonVuLnDetectorTest.java

```diff
@@ -0,0 +1,185 @@
+                .build(),
+            new SlurmExposedRestApiDaemonDetectorBootstrapModule())
+        .injectMembers(this);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    mockTargetService.shutdown();
+    mockCallbackServer.shutdown();
+  }
+
+  @Test
+  public void detect_whenVulnerable_returnsVulnerability() throws IOException {
+    startMockWebServer(true);
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockSuccessfulCallbackResponse());
+
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+
+    assertThat(detectionReports.getDetectionReportsList())
+        .containsExactly(
+            DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(targetNetworkService)
+                .setDetectionTimestamp(
+                    Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                    Vulnerability.newBuilder()
+                        .setMainId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("TSUNAMI_COMMUNITY")
+                                .setValue("SlurmRestApiDaemonRce"))
+                        .setSeverity(Severity.CRITICAL)
+                        .setTitle("Exposed Slurm REST API Server")
+                        .setDescription(
+                            "This detector checks for exposed slurm rest api servers by submitting a job "
+                                + "and checking a callback response on tsunami callback server")
+                        .setRecommendation(
+                            "set proper authentication method for slurm rest api server or "
+                                + "remove the public access to rest api"))
+                .build());
+    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(2);
+    Truth.assertThat(mockCallbackServer.getRequestCount()).isEqualTo(1);
+  }
+
+  @Test
+  public void detect_ifNotVulnerable_doesNotReportVuln() throws IOException {
+    startMockWebServer(false);
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+    assertThat(detectionReports.getDetectionReportsList()).isEmpty();
+    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(2);
+  }
+
+  private void startMockWebServer(boolean withAnExistingModel) throws IOException {
+    final Dispatcher dispatcher =
+        new Dispatcher() {
+          @Override
+          public MockResponse dispatch(RecordedRequest request) {
+            if (Objects.equals(request.getPath(), "/openapi/v3")) {
+              if (request.getMethod().equals("GET")
+                  && request.getBody().readString(StandardCharsets.UTF_8).isEmpty()) {
```

Comment:
```suggestion
            if (Objects.equals(request.getPath(), "/openapi/v3")
                && request.getMethod().equals("GET")
                && request.getBody().readString(StandardCharsets.UTF_8).isEmpty()) {
```
You can AND these conditions instead of using nested ifs